### ISSUE TITLE
Too many log entry

### ISSUE THREAD

**Issue by Darkomen78** [psieg/Lightpack#113]
Anyone can make a "full off log" option ?

When I put Prismatik On, on a 10.12.4 system, I get this line 30 times per second in console :

IOReturn IOAccelSurface2::surface_unlock_options(enum eLockType, uint32_t): surface is not locked.

**Comment by danielcompton** [comment]
This looks like an error log that should be fixed, not ignored?

**Comment by psieg** [comment]
The issue is #93, as a workaround you can make the log file readonly. But I agree with @danielcompton, it sounds like an error message. I assume screen capture works normally for you?
Can any other Mac users reproduce the problem?

**Comment by Darkomen78** [comment]
Sure it's an error message, but anything else seems to work fine.

**Comment by psieg** [comment]
Did this happen before the update too?

**Comment by Darkomen78** [comment]
Yes this happen before the last update and with the "6.0" official version.

**Comment by psieg** [comment]
In a quick search I found no such function call in Prismatik. Does the link danielcompton posted apply to you? Did you change screenshot settings, do you have a touch bar?

**Comment by Darkomen78** [comment]
No change in screenshot settings and no touchbar. No TeamViewer either :-) It's a vanilla-like Sierra install.

**Comment by danielcompton** [comment]
I should have been a little less cryptic in my link. What I suspect is happening is that in a recent OS update, they have changed part of the API contract, and the surface needs to be 'locked'. Whatever we're doing, Teamviewer and the screenshot taking API when it captures to memory are also doing the 'wrong' thing. I'll post in the Apple Developer forums to see if anything comes up there.

**Comment by danielcompton** [comment]
The link is at https://forums.developer.apple.com/message/226018#226018 but it's currently being moderated and I think you'll need a (free) Apple Developer account to read it.

**Comment by psieg** [comment]
Thanks, we'll see how it develops

**Comment by jrnwndr** [comment]
I've been looking into this, and I've discovered a few weird things. If I use VLC to play a video and fullscreen it, Prismatik works perfectly smoothly. Taking screenshots with CMD+Shift+Ctrl+3 doesn't trigger the "surface is not locked" error in the Console either. Maybe we should look into what VLC does when it goes into fullscreen, and apply that to the Prismatik code?

**Comment by psieg** [comment]
I don't see how that would help, as VLC produces screen content and Prismatik consumes it.
When does the message appear on your system?

**Comment by jrnwndr** [comment]
The message appear's in the Mac's Console app. The error is produced when running lightpack, using teamviewer, or taking screenshots. Basically every screen capture software.

Why I'm talking about VLC is because the error's don't exist when I'm using VLC in full screen. Teamviewer is smooth, taking screenshots doesn't produce errors in the Console, and most importantly, Prismatik works perfectly.

Using any other video player fullscreen doesn't help with this, as they use MacOS's native fullscreen function. I'm sure it has to do with the way VLC goes fullscreen on Mac.

**Comment by psieg** [comment]
You say the errors show up if prismatik is on,, but you say the errors don't show up if VLC is fullscreen even if prismatik is on?

**Comment by psieg** [comment]
my guess is this: the error occurs only if an app is fullscreen, but not so for VLC - correct?
Please be as precise as possible to help narrow down the problem.
Though if teamviewer has this problem too, I don't think there's anything we can do

**Comment by danielcompton** [comment]
I've taken a little look at this a while ago, and I think there is something much deeper down in macOS that has changed which is causing screengrabs via this API to log these warnings. Lots of other software (including built-in screenshotting) triggers these warnings. There may be a better API to use than this, but I don't know what that would be.

**Comment by jrnwndr** [comment]
The "surface not locked" error always comes up when using any form of screen capture. Be it Prismatik, Teamviewer or the built in screenshot command. The only time the error doesn't show up is when VLC is fullscreen. Prismatik works great in there.
